bioblend/_tests/TestGalaxyInvocations.py

```diff
@@ -49,16 +66,8 @@ def test_workflow_scheduling(self):
         )
         invocation_id = invocation["id"]
 
-        def invocation_steps_by_order_index():
-            invocation = self.gi.invocations.show_invocation(invocation_id)
-            return {s["order_index"]: s for s in invocation["steps"]}
-
-        for _ in range(20):
-            if 2 in invocation_steps_by_order_index():
-                break
-            time.sleep(.5)
+        steps = self._wait_invocation(invocation_id)
 
```

Comment:
@rikeshi I think you moved the wrong `for` loop to `_wait_invocation()`:

```suggestion
        def invocation_steps_by_order_index():
            invocation = self.gi.invocations.show_invocation(invocation_id)
            return {s["order_index"]: s for s in invocation["steps"]}

        for _ in range(20):
            if 2 in invocation_steps_by_order_index():
                break
            time.sleep(.5)
        steps = invocation_steps_by_order_index()
```

Reply:
That was the only for loop, or not? The removed line and added lines in your suggestion are equivalent since `_wait_invocation()` wraps all of it and returns the steps at the end. Maybe what I did is not what you had in mind. Could you clarify?

Reply:
Oh, I see now. I will fix it!